pandas/core/arrays/integer.py

```diff
@@ -184,6 +184,8 @@ def coerce_to_array(
     -------
     tuple of (values, mask)
     """
+    values = [] if values is np.nan else values
```

Comment:
Can you please look at the caller? This indicates that we're passing `np.nan` to a place where we shouldn't be (probably `IntegerArray._from_sequence`). That means there may be other ExtensionArrays facing the same issue. I'd much rather fix it at the source.

Reply:
> That means there may be other ExtensionArrays facing the same issue

Will we work this on other issues?

Reply:
Depends on the size of the required changes to get this working.

I'm not comfortable merging this until the problem is better understood. We should not be passing `np.nan` to `_from_sequence`. We should be passing `[]`.

Reply:
making changes here may not be necessary once the changes to sanitize_array in #33846 are merged.

Reply:
`pd.DataFrame(columns=["a"], dtype="Int64").dtypes` now works on master following #33846. have reverted this change.